src/Calendar.js

```diff
@@ -56,7 +54,9 @@ const weekNumberCalculation = date => {
 }
 
 export default function Calendar({ timetableState }) {
-  let [eventSources, setEventSources] = useState([])
+  // Ensure we have data
+  if (!timetableState.selectedModules) return <></>
+  if (Object.keys(timetableState.timetableData).length === 0) return <></>
```

Comment:
This means the calendar widget doesn't load for a couple seconds which is a little jarring - could we just use `events = []` if there's no data?

Reply:
Does this work instead?